src/SharethisManager.php

```diff
@@ -83,23 +83,26 @@ public function getOptions() {
    * {@inheritdoc}
    */
   function blockContents() {
+
     $sharethisConfig = $this->configFactory->get('sharethis.settings');
     $config = $this->configFactory->get('system.site');
     if ($sharethisConfig->get('location') == 'block') {
       // First get all of the options for the sharethis widget from the database:
       $data_options = $this->getOptions();
-      $current_path = $_GET['q'];
-      $path = isset($current_path) ? $current_path : '/node';
-      if ($path == $config->get('page.front')) {
-        $path = "node";
-      }
-      $mpath = Url::fromRoute($path, ['absolute' => TRUE]);
-      $mpath = ($mpath->getRouteName());
+      $current_path = \Drupal::routeMatch()->getRouteName() ? Url::fromRouteMatch(\Drupal::routeMatch())->getInternalPath() : '';
```

Comment:
please do inject the current_route_match service in a followup b/c its not a blocker